bare/main.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path"
+	"syscall"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+func cleanup(processes []*os.Process) {
+	for _, p := range processes {
+		p.Kill()
+	}
+	os.Remove("runner")
+}
+
+func bbStart(binaryName, config string) (*os.Process, error) {
+	path, found := bazel.FindBinary(path.Join("cmd", binaryName), binaryName)
+	configPath, err := bazel.Runfile(config)
+	if !found {
+		return nil, fmt.Errorf("Couldn't find %s", binaryName)
+	}
+	if err != nil {
+		return nil, err
+	}
+	cmd := exec.Command(path, configPath)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Process, cmd.Start()
+}
+
+func main() {
+	os.Mkdir("build", 0755)
+	os.Mkdir("cache", 0755)
+	os.Mkdir("storage-ac", 0755)
+	os.Mkdir("storage-cas", 0755)
+
+	bbProcesses := []*os.Process{}
+	bbs := map[string]string{
+		"bare/config/storage.jsonnet":   "bb_storage",
+		"bare/config/frontend.jsonnet":  "bb_storage",
+		"bare/config/scheduler.jsonnet": "bb_scheduler",
+		"bare/config/worker.jsonnet":    "bb_worker",
+		"bare/config/runner.jsonnet":    "bb_runner",
+	}
+	for config, binary := range bbs {
+		p, err := bbStart(binary, config)
+		if err != nil {
+			cleanup(bbProcesses)
+			log.Fatal(err)
+		}
+		bbProcesses = append(bbProcesses, p)
+	}
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
Is all of this really needed? Aren't all of these programs started in the same process group as this launcher tool?